cwl/src/main/scala/cwl/model.scala

```diff
@@ -1,17 +1,44 @@
 package cwl
 
+import cats.data.NonEmptyList
 import eu.timepit.refined._
+import cats.syntax.either._
 import shapeless.{:+:, CNil, Witness}
 import shapeless.syntax.singleton._
 import cwl.LinkMergeMethod.LinkMergeMethod
 import cwl.WorkflowStepInput.InputSource
+import lenthall.validation.ErrorOr.ErrorOr
+import wdl.types.WdlType
+import wom.graph.WomIdentifier
+import wom.graph.GraphNodePort.OutputPort
+import wom.graph.expression.ExposedExpressionNode
 
 case class WorkflowStepInput(
   id: String,
   source: Option[InputSource] = None,
   linkMerge: Option[LinkMergeMethod] = None,
   default: Option[CwlAny] = None,
-  valueFrom: Option[StringOrExpression] = None)
+  valueFrom: Option[StringOrExpression] = None) {
+
+  def toExpressionNode(sourceMappings: Map[String, OutputPort],
+                       outputTypeMap: Map[String, WdlType],
+                       inputs: Set[String]
+                      ): ErrorOr[ExposedExpressionNode] = {
+    val source = this.source.flatMap(_.select[String]).get
```

Comment:
We'll have to do a sweep for those `get`s at some point or I predict a bunch of "My workflow fails with None.get" tickets popping up when we'll release CWL 😄